refactor: allow managed event tests parallel runs

zomars opened the pull request:
## What does this PR do?

Refs #16099

- Removed low quality test: `apps/web/playwright/managedBooking/advancedOptions.e2e.ts` as it was only checking the offer seats toggle without even testing booking page functionality. Removed until we come up with something better.
- Split managed event type tests to run in parallel instead on sequence. This should improve overall testing time.
These tests seems to be borked in main as well. Here's [an example ](https://github.com/calcom/cal.com/actions/runs/10271190062/job/28420916680)of them timing out but still are marked as "passed"
- Added `managedEventUnlockedFields` parameter to the `createTeamEventType` function to allow specifying unlocked fields for managed events.
- Updated `createUsersFixture` to include `managedEventUnlockedFields` parameter and logic to handle adding teammates to managed events.
- Enhanced related e2e tests in `managed-event-types.e2e.ts` to validate functionality around managed events including locked/unlocked fields, organizer default apps, and tab renders.

### Before:

<img width="372" alt="image" src="https://github.com/user-attachments/assets/e3fff546-4c25-403f-a731-64931ff12c9b">


<img width="1001" alt="image" src="https://github.com/user-attachments/assets/3222057f-5c1e-4555-917e-07857b705773">

### After

<img width="363" alt="image" src="https://github.com/user-attachments/assets/8cf5adb4-47f4-40ee-8b7a-10c2742f6cd6">

<img width="1050" alt="image" src="https://github.com/user-attachments/assets/bff5b01f-49d6-4449-b1f0-ba0c5e78ab0a">

### How to test?

1. Execute e2e tests in `managed-event-types.e2e.ts` to ensure that managed event type functionalities are correctly implemented.

---

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=16113) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=16113) before we can accept your contribution.<br/>**12** out of **17** committers have signed the CLA.<br/><br/>:white_check_mark: keithwillcode<br/>:white_check_mark: joeauyeung<br/>:white_check_mark: Udit-takkar<br/>:white_check_mark: hbjORbj<br/>:white_check_mark: anikdhabal<br/>:white_check_mark: ThyMinimalDev<br/>:white_check_mark: Souptik2001<br/>:white_check_mark: supalarry<br/>:white_check_mark: darshil89<br/>:white_check_mark: thepradipvc<br/>:white_check_mark: ImBIOS<br/>:white_check_mark: SySagar<br/>:x: nebulade<br/>:x: maxprilutskiy<br/>:x: nicmosc<br/>:x: calcom-bot<br/>:x: pietrzakacper<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=16113) it.</sub>

zomars committed:
refactor: allow managed event tests parallel runs

commit a89097842bfceffdde998a13a24eb17bb0952a89
Author: zomars <zomars@me.com>
Date:   Wed Sep 4 16:18:15 2024 -0700

    Update managed-event-types.e2e.ts

commit 0f6d3abb756f8abf14a66e42ae68c9bded8c71a4
Merge: 3dc385b4fe 36e79bf69c
Author: zomars <zomars@me.com>
Date:   Wed Sep 4 14:48:28 2024 -0700

    Merge branch 'main' into refactor_allow_managed_event_tests_parallel_runs

commit 3dc385b4feeafe95c0b3fb6a4bfbfd424327fc71
Author: zomars <zomars@me.com>
Date:   Wed Sep 4 11:48:24 2024 -0700

    Update users.ts

commit 52289d30ab66eb61bc27b1bb851b5b34fbf3deac
Merge: e04ecd9080 e9c71abd6f
Author: zomars <zomars@me.com>
Date:   Wed Sep 4 11:38:38 2024 -0700

    Merge branch 'main' into refactor_allow_managed_event_tests_parallel_runs

commit e04ecd9080f2a0f12f80e40583d727abbcc0c917
Author: zomars <zomars@me.com>
Date:   Tue Aug 6 23:25:45 2024 -0700

    refactor: allow managed event tests parallel runs

commit 96c2349f660c48725ef2f7d34aa530c9fb6e1d48
Author: Carina Wollendorfer <30310907+CarinaWolli@users.noreply.github.com>
Date:   Wed Sep 4 13:56:20 2024 -0400

    fix: meeting url missing in workflow email (#16434)

    Co-authored-by: CarinaWolli <wollencarina@gmail.com>
    Co-authored-by: Joe Au-Yeung <65426560+joeauyeung@users.noreply.github.com>

commit 03ef1ff0de0f36dca9eddb9bcfc5438b578fdddc
Author: Imamuzzaki Abu Salam <imamuzzaki@gmail.com>
Date:   Wed Sep 4 23:02:30 2024 +0700

    chore: Remove obsolete docker compose version (#16419)

commit 0b9962a08a83656e6f2525da9360d00f069b17a1
Author: Benny Joo <sldisek783@gmail.com>
Date:   Wed Sep 4 11:20:40 2024 -0400

    chore: App router migration - Add env variables for `auth/forgot-password/*`, `auth/login`, `auth/logout`, `auth/new`, `auth/saml-idp (#16488)

    * Add env variables for auth/forgot-password, auth/login, auth/logout

    * Add env variables for auth/new, auth/saml-idp

commit 91813b9491341796c971e15e4882665417c5fd84
Author: Morgan <33722304+ThyMinimalDev@users.noreply.github.com>
Date:   Wed Sep 4 16:56:00 2024 +0300

    fix: api v2 booking controller handle error codes (#16486)

    * fix: api v2 booking controller handle error codes

    * fixup! fix: api v2 booking controller handle error codes

    * fixup! Merge branch 'main' into fix-booking-controller-errors-api-v2

commit 2ada394f4b84d4dfed5d68af4ab30df3211e039c
Author: Benny Joo <sldisek783@gmail.com>
Date:   Wed Sep 4 09:15:19 2024 -0400

    chore: App-router-migration `/auth/error`, `/auth/platform`, `/auth/oauth2` (#16445)

    * preparePageMetadata: improve logic for title

    * add missing pages: oauth2 / platform

    * auth/error: add to app router and extract to /modules

commit b93b578771045760bdde8faa74c9e70dd8512729
Author: Benny Joo <sldisek783@gmail.com>
Date:   Wed Sep 4 09:05:38 2024 -0400

    chore: App-router-migration `/auth/new` and `/auth/saml-idp` (#16447)

    * auth/new: add to app router

    * auth/saml-idp: add to app router

    * fix

commit f3b4776e7de4f3d709e9bd010adc583acf040a5d
Author: Lauris Skraucis <lauris.skraucis@gmail.com>
Date:   Wed Sep 4 13:17:13 2024 +0200

    fix: use dbWrite in OrganizationsTeamsRepository (#16485)

commit ab16968aadc18dc4cdbd9dc5e3e6a8ab3821b748
Author: Lauris Skraucis <lauris.skraucis@gmail.com>
Date:   Wed Sep 4 13:14:05 2024 +0200

    chore: v2 bump platform libraries (#16473)

commit 234701d5fb4bb1b829b7186bf15f2059b5a3608a
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Wed Sep 4 08:09:10 2024 -0300

    chore: Remove unnecessary log (#16426)

commit d1361dc1245277058e77a0a7a77cad28b199f23b
Author: Peer Richelsen <peeroke@gmail.com>
Date:   Wed Sep 4 13:05:43 2024 +0200

    Update bug_report.yml (#16089)

commit 20981874d75e240d1cb0e42746868ed4b9fa73f8
Author: Calcom Bot <109866826+calcom-bot@users.noreply.github.com>
Date:   Wed Sep 4 12:39:16 2024 +0200

    feat: update translations (#16477)

    Co-authored-by: Replexica <support@replexica.com>

commit 8603bff5cbdf8d40204e0f742d0fcca726c840be
Author: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>
Date:   Tue Sep 3 19:58:15 2024 +0530

    feat: availability in instant meeting  (#16424)

    * chore: save progress

    * feat: add isAvailable functionality

    * fix: type error

    * chore: add in builder

    * tests: add unit tests

    * chore: improvements

    * chore: tests

    * chore

    * chore: remove log

    * fix: tets

    ---------

    Co-authored-by: Morgan <33722304+ThyMinimalDev@users.noreply.github.com>

commit 90bf0833a79356b30cc15326e09e3ffe33608c29
Author: Darshil Mharaur <109381405+darshil89@users.noreply.github.com>
Date:   Tue Sep 3 19:04:05 2024 +0530

    fix: Adding yearly title to preview page (#16441)

    * initial commit

    * added year tag to the occurences field

    * Revert yarn.lock

    ---------

    Co-authored-by: Amit Sharma <74371312+Amit91848@users.noreply.github.com>

commit fcce3296d8fe6bde81548e79ad121edffe439a31
Author: Calcom Bot <109866826+calcom-bot@users.noreply.github.com>
Date:   Tue Sep 3 15:20:21 2024 +0200

    feat: update translations (#16475)

    Co-authored-by: Replexica <support@replexica.com>

commit 022410c9c136cdb7acddfb410b25ecf7a2a40a9e
Author: Hariom Balhara <hariombalhara@gmail.com>
Date:   Tue Sep 3 18:28:34 2024 +0530

    test: Unit Tests for FormBuilderField and BookingFields components (#16162)

    * Remove use of location from FormBuilder

    * Add tests

    * FormBuilderField and BookingFields tests

    * More tests

    * Remove always true if condition

    * Fix ui import mockig that got broken after the last merge

commit cd3745dc65e5659d0c1f022404fffbebabb61de3
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Tue Sep 3 14:34:50 2024 +0200

    chore: Add 3 more languages to AI translations (#16469)

commit d96f17ce8abb0d8b5432b30738cc5c9fdec6f6d1
Author: Benny Joo <sldisek783@gmail.com>
Date:   Tue Sep 3 05:53:16 2024 -0400

    chore: App router migration - `/auth/forgot-password/*`, `/auth/login`, `/auth/logout` (#16467)

    * migrate auth/forgot-password/*, auth/login, auth/logout

    * revert

    * fix forgot password

    * fix folder structure

    * fix logout

    * fix middleware

    * remove TODO log

    * refactor logout

commit dda91117f57a599e3691460e7e9d5394a0ed4022
Author: Benny Joo <sldisek783@gmail.com>
Date:   Tue Sep 3 05:53:02 2024 -0400

    chore: App router migration - Add method to `BookingRepository` & Extract `/videos/**/*` into `/modules` and `/lib` (#16466)

    * meeting-ended: refactor and extract client code into /modules

    * meeting-not-started: refactor and extract client code into /modules

    * meeting-not-found: fix getData

    * BookingRepository: add a new method

    * video: refactor and extract client code into /modules

    * refactor

    * refactor getServerSideProps

    * use findBookingByUid for more getServerSideProps functions

    * rename findBookingByUid -> findBookingByUidWithOptionalSelect

    * add safety

commit d1be02fa9754d5cd6ea8ff031737eee35cec3a20
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Tue Sep 3 08:54:12 2024 +0200

    chore: Log cold starts of /v1/slots (#16468)

commit b90e0de01991360fa47cb105fa5bd6c37d2fb1b4
Author: Calcom Bot <109866826+calcom-bot@users.noreply.github.com>
Date:   Tue Sep 3 08:43:44 2024 +0200

    feat: update translations (#16462)

    Co-authored-by: Replexica <support@replexica.com>

commit 1b9e3450a961be0f9ff7d86d7f8267e03163b200
Author: Benny Joo <sldisek783@gmail.com>
Date:   Mon Sep 2 13:30:48 2024 -0400

    chore: App router migration (`/workflows`) fix metadata and remove duplicate file for `getStaticProps` (#16452)

    * Fix metadata

    * add use client directive to workflow component

commit ec461f3c10f1d01a7cbbd401d1b413445f789853
Author: Pradip Chaudhary <88336529+thepradipvc@users.noreply.github.com>
Date:   Mon Sep 2 20:09:51 2024 +0530

    feat: browser push notifications (#14888)

    * feat: web push notifications feature

    * Revert yarn.lock changes

    * added new env variables requirement in .env.example

    * moved useNotifications hook in packages/lib/hooks

    * fix: bug

    * use i18n

    * chore: move to new file

    * chore: add yarn.lock

    * Update .env.example

    Co-authored-by: Amit Sharma <74371312+Amit91848@users.noreply.github.com>

    * chore: add instruction for brave browser

    * fix: tooltip

    * chore: use enum

    * chore

    * small update

    ---------

    Co-authored-by: Peer Richelsen <peeroke@gmail.com>
    Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>
    Co-authored-by: Udit Takkar <udit222001@gmail.com>
    Co-authored-by: Peer Richelsen <peer@cal.com>
    Co-authored-by: Amit Sharma <74371312+Amit91848@users.noreply.github.com>
    Co-authored-by: Carina Wollendorfer <30310907+CarinaWolli@users.noreply.github.com>
    Co-authored-by: unknown <adhabal2002@gmail.com>
    Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>

commit 39ffe6bf8098af57fb3e7e9aebf7c99598e0c057
Author: Hariom Balhara <hariombalhara@gmail.com>
Date:   Mon Sep 2 16:12:11 2024 +0530

    Fix unconfirmed bookings blocking slots being fetched for all troubleshooters (#16439)

    Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

commit 35b5dac3adb446d27cb5a9ea9e5b4f5f5f8ac3d8
Author: Calcom Bot <109866826+calcom-bot@users.noreply.github.com>
Date:   Sat Aug 31 17:17:34 2024 +0200

    feat: update translations (#16436)

    Co-authored-by: Replexica <support@replexica.com>

commit 5fc80850473fd0e748cf0c140717b21175137401
Author: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>
Date:   Sat Aug 31 02:08:09 2024 +0530

    feat: add "somewhere else" as a location option (#15108)

    * feat: add somewhere else as a location

    * update

    * update

    * Update BookingFields.tsx

    * update

    * update

    * chore: changes

    * fix

commit 33d14575aa83cd67d7de40f36ea52733be10ced0
Author: Peer Richelsen <peeroke@gmail.com>
Date:   Fri Aug 30 18:38:01 2024 +0200

    feat: synthflow app (#16420)

commit 102c6a5754e69b0bb14443ab8b8ba3c27d9af7e8
Author: Kacper <kacpietrzak@gmail.com>
Date:   Fri Aug 30 18:07:10 2024 +0200

    Fallback to next app default language (en) when user didn't specify locale. (#16425)

commit e0989c34fbfa76e70cc70aeea3e1bbc1a3c0570a
Author: Lauris Skraucis <lauris.skraucis@gmail.com>
Date:   Fri Aug 30 14:07:53 2024 +0200

    fix: recurring booking idempotency key after cancel (#16414)

    * fix: recurring booking idempotency key after cancel

    * refactor: move logic to updateMany of prisma middleware

commit b5969228b8bd1067fba3541853a5e4cee55fe666
Author: Morgan <33722304+ThyMinimalDev@users.noreply.github.com>
Date:   Fri Aug 30 13:41:37 2024 +0300

    chore: reorganize event type files (#16393)

    * chore: reorganize event type files

    * fixup! chore: reorganize event type files

    * fixup! Merge branch 'main' into refactor-event-types-components-folder

commit 51d210e9e11d5a25fceebad7a293c2c46ba29d70
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Fri Aug 30 19:10:44 2024 +0900

    perf: Improve loading existing bookings via API (#16410)

commit 143ff408b8ab6d52af8e032784fcf5a5650f1563
Author: Calcom Bot <109866826+calcom-bot@users.noreply.github.com>
Date:   Fri Aug 30 01:26:01 2024 +0200

    feat: update translations (#16401)

    Co-authored-by: Replexica <support@replexica.com>

commit d54dd56ecdbe65c58c5c953fb321b1d6d8136e71
Author: Hariom Balhara <hariombalhara@gmail.com>
Date:   Fri Aug 30 03:05:32 2024 +0530

    fix: Broken reschedule page for migrated user's booking through request-reschedule (#16068)

    Co-authored-by: Syed Ali Shahbaz <52925846+alishaz-polymath@users.noreply.github.com>
    Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>
    Co-authored-by: Peer Richelsen <peeroke@gmail.com>
    Co-authored-by: Amit Sharma <74371312+Amit91848@users.noreply.github.com>
    Co-authored-by: zomars <zomars@me.com>

commit bbde6b168e5d4ecbc62bc4f62297d0de0fcb1b2d
Author: Omar López <zomars@me.com>
Date:   Thu Aug 29 14:26:05 2024 -0700

    chore: v4.4.5 (#16402)

commit af7347ec7f3faa42db9d1f9dbd22ad9f5613018d
Author: Joe Au-Yeung <65426560+joeauyeung@users.noreply.github.com>
Date:   Thu Aug 29 17:16:48 2024 -0400

    fix: Set `teamMemberEmail` server side for CRM RR Skip (#16367)

    Co-authored-by: Keith Williams <keithwillcode@gmail.com>
    Co-authored-by: zomars <zomars@me.com>

commit 45c3724db9f69f35bba2583b70a48cfc27054e4e
Author: Benny Joo <sldisek783@gmail.com>
Date:   Thu Aug 29 17:03:02 2024 -0400

    chore: App Router Migration - Ensure `isBookingPage` is correctly flagged before being passed to `PageWrapper` (#16362)

    * Ensure adding isBookingPage is done in components in /modules, not in /pages

    * Detect isBookingPage prop coming from Page and add it to page wrapper app dir

    * fix build

    * fix

    ---------

    Co-authored-by: Joe Au-Yeung <65426560+joeauyeung@users.noreply.github.com>

commit add80d87be4488d71c4f89ffc83f538a619c95be
Author: Hariom Balhara <hariombalhara@gmail.com>
Date:   Fri Aug 30 02:24:41 2024 +0530

    fix: Add version to embed.js and preview.js (#16182)

commit e81cc38a722b7fda8f46c01ceb970805f8eb5b73
Author: Hariom Balhara <hariombalhara@gmail.com>
Date:   Fri Aug 30 01:44:12 2024 +0530

    fix: Flicker b/w Dark and Light theme (#16042)

commit a0c55e704537b2101025767e180378514caae7dc
Author: Souptik Datta <souptikdatta2001@gmail.com>
Date:   Thu Aug 29 22:58:35 2024 +0530

    feat: Implement editable OOO events feature (#15932)

    * feat: Implement edit ooo feature

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * refactor: Refactor button text and gap between buttons

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * fix: Fix entry create and edit edge cases

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * chore: Change "edit ooo event" modal title

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * chore: Add tooltips to entry eidt and delete buttons

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * feat: Reset form on form close event

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * refactor: Refactor reset-form function

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * chore: Rename create or edit OOO as suggested

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * refactor: Refactor component and function names

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * feat: Improve ooo events email notifications

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * fix: Fix email templates

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * feat: Add e2e test for ooo event edit functionality

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * refactor: Create separate test for edit ooo event

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * fix: Remove extra user from redirect user configuration test

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * chore: code refactor

    * remove log

    * chore: Update ooo-input-schema uuid data type

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * fix: Address feedbacks

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

    * chore: code splitting and refactor

    ---------

    Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>
    Co-authored-by: Amit Sharma <74371312+Amit91848@users.noreply.github.com>
    Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>

commit ec98706aeab4bbecfcb1a1819cdd1256f0eb67ae
Author: Calcom Bot <109866826+calcom-bot@users.noreply.github.com>
Date:   Thu Aug 29 19:13:06 2024 +0200

    feat: update translations (#16400)

    Co-authored-by: Replexica <support@replexica.com>

commit a96075b2b038682c34ac0bdef21558ec7716f5d4
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Fri Aug 30 01:37:18 2024 +0900

    perf: run parallel bookings queries (#16398)

commit 056652e47603a6a0c68e9139474102e4f45a5a94
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Fri Aug 30 01:18:48 2024 +0900

    feat: Add more AI translation languages (#16381)

    * feat: Add more AI translation languages

    * Removed pull-request: true

    * checking out the PRs i18n.json as well

    * Update i18n.json

    * Update i18n.json

commit 9ccff3bdf7a55a64b2a986c9a9a6567751a88404
Author: Calcom Bot <109866826+calcom-bot@users.noreply.github.com>
Date:   Thu Aug 29 17:58:25 2024 +0200

    feat: update translations (#16396)

    Co-authored-by: Replexica <support@replexica.com>

commit 230366c79840be4f5543d82f82a909f96b291430
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Fri Aug 30 00:02:05 2024 +0900

    fix: GitHub access token for Replexica (#16395)

commit 40c9330218e9f5568dd369e13a7f141458b7ac99
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Thu Aug 29 23:22:42 2024 +0900

    chore: Move Replexica to run on pushes to main (#16394)

commit 37b7c7f588474bd69ea886be138bac0eac0c4947
Author: Syed Ali Shahbaz <52925846+alishaz-polymath@users.noreply.github.com>
Date:   Thu Aug 29 17:52:42 2024 +0400

    feat: Org Webhooks API V2 (#16274)

    * --init

    * add is webhook in org guard

    * --

    * doc

    * e2e

    * --feedback

    * --

    * fix typo

    * delete unnecessary file

    * fix v2 e2e

    * fix v2 e2e

    * fix e2e

    * fix v2 e2e

    ---------

    Co-authored-by: Morgan <33722304+ThyMinimalDev@users.noreply.github.com>

commit 925558bf3c5136e4d9496d4bf5d084479cff6f52
Author: Sy Sagar <93787385+SySagar@users.noreply.github.com>
Date:   Thu Aug 29 17:46:13 2024 +0530

    chore: Bulk update desc (#16343)

    * chore(locale-en): update bulk modal description

    * chore: schedule bulk-modal description

    ---------

    Co-authored-by: Amit Sharma <74371312+Amit91848@users.noreply.github.com>

commit 981b01784e93227036e6ca34fbffb8d9eec22207
Author: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>
Date:   Thu Aug 29 13:32:54 2024 +0530

    refactor: improve team members page performance (#16155)

    * fix: simplify workflow page and improve load time

    * chore: use new endpoint

    * chore: save progress

    * refactor: code

    * refactor: remove not requried code

    * chore: remove schema

    * chore: fix typ

    * chore: improve

    * chore: change name

    * chore: remove unused

    * chore: remove page

    * refactor: teams page

    * feat: add auto scroll

    * chore: create validate unique invite

    * fix: auth check

    * fix: optimistic update

    * chore

    * fix: add loading

    * fix: improvements

    * chore: remove

    * chore

    * chore: fix teams page

    * fix: team profile page

    * fix: appearance page

    * fix: sso view

    * fix: type err

    * feat: defer loading connected Apps

    * fix: type err

    * fix: type error

    * fix: type err

    * fix: connectedApps type

    * chore: move

    * chore: missing export

    * feat: add search by name

    * fix: display role change

    * fix: use setInfiniteData

    * chore: save progress

    * test: add unit tests for loading members

    * fix: test

    * chore: update name

    * fix: bugs and improvements

    * chore: change variable name

    * test: add tests for checkCanAccessMembers

    * refactor: performance

    ---------

    Co-authored-by: Joe Au-Yeung <65426560+joeauyeung@users.noreply.github.com>

commit 3f7cc2eb271b5fd880c5f053e68f07cca7dd50d2
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Thu Aug 29 14:13:48 2024 +0900

    perf: Removed unused attendees count (#16391)

commit 40a611644fa2fecffdcdab49615744550048a359
Author: Max Prilutskiy <5614659+maxprilutskiy@users.noreply.github.com>
Date:   Thu Aug 29 03:32:03 2024 +0200

    chore: [skip i18n] switch pr's i18n job to in-branch mode (#16390)

commit 6eecbf08b067e42188f932a9128185bcc2fa8936
Author: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>
Date:   Wed Aug 28 20:54:13 2024 +0530

    fix: allow all sign in methods when being invited (#13436)

    * allow all sign in methods

    * changes

    * update

    * add checks

    ---------

    Co-authored-by: Keith Williams <keithwillcode@gmail.com>
    Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>
    Co-authored-by: Joe Au-Yeung <65426560+joeauyeung@users.noreply.github.com>

commit 8c4779a4b172fa88193bb6fa46156ddef37ec4d7
Author: Nicolaos Moscholios <nmoscholios@gmail.com>
Date:   Wed Aug 28 15:37:02 2024 +0200

    fix: use `userPhone` instead of `phone` for switch case (#16384)

    * fix: use `userPhone` instead of `phone` for switch case

    Currently this switch case never gets to the last case ("phone") as the type expected is wrong.

    Here the type may have value `userPhone` and not `phone`.

    * fix another case

    ---------

    Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>

commit d315dfd1efa8dd55bb43c5106e5ec844b853c65e
Author: Morgan <33722304+ThyMinimalDev@users.noreply.github.com>
Date:   Wed Aug 28 16:05:14 2024 +0300

    fix: api v2 booking controller with api key (#16358)

    * fix: api v2 booking controller with api key

    * add test

    * fixup! add test

    * fixup! Merge branch 'fix-booking-controller-api-key' of github.com:calcom/cal.com into fix-booking-controller-api-key

    * fixup! Merge branch 'fix-booking-controller-api-key' of github.com:calcom/cal.com into fix-booking-controller-api-key

commit d6ba39159f88b1079031d82eaaa56cf3979c7ed1
Author: Joe Au-Yeung <65426560+joeauyeung@users.noreply.github.com>
Date:   Wed Aug 28 08:53:32 2024 -0400

    fix: Move `future/workflow/[workflow]` `getStaticProps` into separate file (#16124)

commit a0c99434e8bb074dfff3c40712d8185b3d54321d
Author: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>
Date:   Wed Aug 28 18:01:06 2024 +0530

    chore: display first 3 organizers photo in round robin event (#16331)

    * chore: display random organizer photo in round robin event

    * fix: avatar on instant booking

    ---------

    Co-authored-by: sean-brydon <55134778+sean-brydon@users.noreply.github.com>

commit 10c2c63075aeea9c7071ee6f67ffbbd2b4b19e22
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Wed Aug 28 19:24:56 2024 +0900

    chore: Use pull-request feature from Replexica (#16383)

commit 989d5b3ac13d4285d1622232cf222f0559b363bd
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Wed Aug 28 17:16:47 2024 +0900

    chore: add replexica to ci (#16306)

    * chore: Add replexica for AI i18n

    * Add write permission

    * moved i18n to repo root

    * Added checkout to i18n job

    * Added contents:read

    * Added actions: write for testing

    * Updated contents: write

    * feat: add missing translations

    * Checking out the i18n files from PR head sha

    * Added the rest of the languages

    * Added i18n to list of required checks

    * Added a comment explaining the custom common.json checkout

    * Commented out languages to not crush Replexica usage

    * Add a fetch-depth: 2

    * Supporting fetch of external fork branch

    * Put back the dynamic sha

    * Added a check for null so this can run manually and not just on PRs

    * Removed comments

    * feat: update translations

    * Reverted de and fr to re-run with new settings

    * feat: update translations

    * Reverted de and fr to re-run with new settings

    * feat: update translations

    * Removed crowdin

    * Limited languages to major ones for 1st pass

    * feat: update translations

    * Fixing es test

    * Reverted yarn.lock changes

    ---------

    Co-authored-by: Replexica <support@replexica.com>

commit 73ecade22925c8d8ff0d72cd815c207adc24b0d2
Author: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>
Date:   Wed Aug 28 04:10:00 2024 +0530

    fix: require confirmation cal video bug (#16376)

    * fix: require confirmation cal video bug

    * chore: update location

    * fix: tests

    * chore

    * test: add confirm booking test

    * chore: add location in test

    * fix: types

commit c1fa29545511dd0730b14a71724883bbde472688
Author: sean-brydon <55134778+sean-brydon@users.noreply.github.com>
Date:   Tue Aug 27 23:30:18 2024 +0100

    use goblin url (#16370)

commit fa65b4db8d73fa8f1b5cc6c984e72f9805b5a64f
Author: Benny Joo <sldisek783@gmail.com>
Date:   Tue Aug 27 13:28:09 2024 -0400

    chore: App Router Migration - Fix metadata util used in App Router and remove duplicate code (#16346)

commit fd2be02f1022623300eb6c6343b0569f6f190d84
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Tue Aug 27 22:38:34 2024 +0900

    chore: v4.4.4 (#16372)

commit ca718ad56d91f715a593b949932f6590f7d60194
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Tue Aug 27 21:25:53 2024 +0900

    fix: API v2 E2E Docker rate limiting (#16374)

commit f837e048e77a6b61bcf667d79c3085900d9af3a0
Author: Hariom Balhara <hariombalhara@gmail.com>
Date:   Tue Aug 27 17:44:19 2024 +0530

    fix: More Stability for FormBuilder by unit tests (#16129)

    * Remove use of location from FormBuilder

    * Add tests

    * Improve error message in test

    * Bust cache

commit 705898b5a4b2a5bc6d897020e3a389c1f4382a08
Author: Johannes Zellner <johannes@nebulon.de>
Date:   Tue Aug 27 09:46:55 2024 +0200

    fix: Avoid typescript error for Nextjs usePathname() (#16368)

    * Avoid typescript error for Nextjs usePathname()

    #fixes 16323

    * Update Shell.tsx

    ---------

    Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>

commit 1dd76391c36ef1b3ad6030a7edc66ce0a3bb3d50
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Tue Aug 27 09:43:04 2024 +0900

    fix: Undefined user in insights data (#16365)

    * fix: Null user in insights data

    * renaming

commit 17692279b1b07c63c97dbf2141984ad04cbdc4f6
Merge: edd299f15b 4caa095d52
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Tue Aug 27 08:02:57 2024 +0900

    Merge branch 'main' into refactor_allow_managed_event_tests_parallel_runs

commit edd299f15b735ea50cfb6d42c55c1def2b267118
Merge: e4e78d4fd2 0b719ac070
Author: Keith Williams <keithwillcode@gmail.com>
Date:   Wed Aug 21 11:38:49 2024 +0900

    Merge branch 'main' into refactor_allow_managed_event_tests_parallel_runs

commit e4e78d4fd2a993189b8b1dfedd54ba7f0c6b0863
Merge: e7c2c8b3a6 cb6553c731
Author: Omar López <zomars@me.com>
Date:   Fri Aug 16 10:41:41 2024 -0700

    Merge branch 'main' into refactor_allow_managed_event_tests_parallel_runs

commit e7c2c8b3a6984f1a25d69c3a69d3d86023d92088
Merge: 55b0b90d7e 08e1b0a9c8
Author: Omar López <zomars@me.com>
Date:   Thu Aug 15 14:51:09 2024 -0700

    Merge branch 'main' into refactor_allow_managed_event_tests_parallel_runs

commit 55b0b90d7e7ab6a90d10a97cd757cfd588a0405a
Merge: 61889abc78 5df14c344e
Author: Omar López <zomars@me.com>
Date:   Thu Aug 15 13:26:28 2024 -0700

    Merge branch 'main' into refactor_allow_managed_event_tests_parallel_runs

commit 61889abc787316122376c6ca756e4b862382e596
Merge: 9ebbc0365e cd311f076c
Author: zomars <zomars@me.com>
Date:   Thu Aug 15 11:53:18 2024 -0700

    Merge branch 'main' into refactor_allow_managed_event_tests_parallel_runs

commit 9ebbc0365e397d7b78a18a332198a490510412e2
Merge: a4f482c588 f206b3d135
Author: zomars <zomars@me.com>
Date:   Tue Aug 13 16:20:25 2024 -0700

    Merge branch 'main' into refactor_allow_managed_event_tests_parallel_runs

    Signed-off-by: zomars <zomars@me.com>

    # Conflicts:
    #	apps/web/playwright/fixtures/users.ts

commit a4f482c588261d9169ff937d59f3d9704844a3e0
Author: zomars <zomars@me.com>
Date:   Mon Aug 12 14:13:05 2024 -0700

    fix tests

    Signed-off-by: zomars <zomars@me.com>

commit e12f0f50003f035a93095c34cdd676b7dc74b64a
Author: zomars <zomars@me.com>
Date:   Tue Aug 6 23:33:59 2024 -0700

    type fix

commit 7c0ef775fa072e062b63115ad62b49f900f1bb2f
Author: zomars <zomars@me.com>
Date:   Tue Aug 6 23:25:45 2024 -0700

    refactor: allow managed event tests parallel runs

zomars committed:
Update managed-event-types.e2e.ts

zomars committed:
Update managed-event-types.e2e.ts

zomars committed:
Merge branch 'main' into refactor_allow_managed_event_tests_parallel_runs

# Conflicts:
#	apps/web/playwright/managed-event-types.e2e.ts

zomars committed:
Update managed-event-types.e2e.ts

zomars committed:
Discard changes to i18n.json

zomars committed:
Discard changes to apps/api/v2/src/modules/organizations/services/attributes/organization-attributes.service.ts

zomars committed:
revert

zomars committed:
Update managed-event-types.e2e.ts

zomars commented on apps/web/playwright/managed-event-types.e2e.ts:
```diff
@@ -1,212 +1,245 @@
+  return { adminUser, memberUser, managedEvent, teamMateName, teamEventTitle };
+}
+
+const getByKey = async (page: Page, key: string) => page.getByText((await localize("en"))(key));
```
Short hand to get elements by translation key

zomars commented on apps/web/playwright/managed-event-types.e2e.ts:
```diff
@@ -1,212 +1,246 @@
-      await page.click("[type=submit]");
-      // TODO: Figure out a way to make this more reliable
-      // eslint-disable-next-line playwright/no-conditional-in-test
-      if (IS_TEAM_BILLING_ENABLED) await fillStripeTestCheckout(page);
```
We shouldn't test for Stripe in here. It's redundant.

zomars commented on apps/web/playwright/managed-event-types.e2e.ts:
```diff
@@ -1,212 +1,246 @@
     // Let's create a team
-    await page.goto("/settings/teams/new");
-
-    await test.step("Managed event option exists for team admin", async () => {
```
We already know if the option exists in the "Can create managed event type" test. Removed due redundancy.

zomars committed:
Update managed-event-types.e2e.ts

zomars commented on apps/web/playwright/managed-event-types.e2e.ts:
```diff
@@ -1,212 +1,247 @@
-
-      await page.getByTestId(`select-option-${memberUser.id}`).click();
-      await page.locator('[type="submit"]').click();
-      await page.getByTestId("toast-success").waitFor();
```
Now we wait for the POST request instead of relying on toast which is very flaky.

zomars commented on apps/web/playwright/managed-event-types.e2e.ts:
```diff
@@ -1,212 +1,247 @@
-      await page.getByTestId("toast-success").waitFor();
-      await page.waitForLoadState("networkidle");
-
-      await page.goto("/auth/logout");
```
No need to logout, we create a new browser to edit stuff as admin/member.

zomars commented on apps/web/playwright/managed-event-types.e2e.ts:
```diff
@@ -1,212 +1,247 @@
+    await memberPage.locator('input[name="title"]').fill(`Managed Event Title`);
+    await submitAndWaitForResponse(memberPage);
+
+    // We edit the managed event as original owner
+    const adminContext = await browser.newContext();
+    const adminPage = await adminContext.newPage();
+    const adminUserSnapshot = await adminUser.self();
+    await apiLogin({ ...adminUserSnapshot, password: adminUserSnapshot.username }, adminPage);
+    await adminPage.goto("/event-types");
+    await adminPage.getByTestId("event-types").locator(`a[title="${teamEventTitle}"]`).click();
+    await adminPage.waitForURL("event-types/**");
+    await adminPage.locator('input[name="length"]').fill(`45`);
+    await submitAndWaitForResponse(adminPage);
+    await adminContext.close();
```
So we don't wait on login/out flows

zomars commented on apps/web/playwright/managed-event-types.e2e.ts:
```diff
@@ -1,212 +1,247 @@
+  ];
+
+  MANAGED_EVENT_TABS.forEach((tab) => {
+    test(`Managed event can render "${tab.slug}" tab`, async ({ page, users }) => {
```
This would have helped catching a previous error where the availability tab stopped rendering properly.

zomars commented on apps/web/playwright/managed-event-types.e2e.ts:
```diff
@@ -216,3 +251,10 @@ async function gotoBookingPage(page: Page) {
   await page.goto(previewLink ?? "");
 }
+
+async function submitAndWaitForResponse(page: Page) {
+  const submitPromise = page.waitForResponse("/api/trpc/eventTypes/update?batch=1");
+  await page.locator('[type="submit"]').click();
+  const response = await submitPromise;
+  expect(response.status()).toBe(200);
+}
```
I found this pattern to be a reliable way to replace waiting on toast-success after saving changes. cc @anikdhabal

anikdhabal replied:
I've noticed that sometimes a reload occurs after a successful save, especially for event types, and the toast message isn't visible. This is why expecting a success toast message can be flaky. I think we can use it in other tests and apply it to other event type update cases as well

zomars reviewed (commented):
Ready for review

zomars committed:
Merge branch 'main' into refactor_allow_managed_event_tests_parallel_runs

zomars committed:
Cleanup

zomars commented on apps/web/playwright/managedBooking/advancedOptions.e2e.ts:
Removed due to low quality test as stated in the PR description.

anikdhabal reviewed (approved):
LGTM!
That's a great workaround! The visible result is undoubtedly top-notch 🚀
